gems/smithy-client/lib/smithy-client/identity_provider.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    # Basic Identity resolver that uses a proc to resolve an Identity.
+    class IdentityProvider
```

Comment:
I'm guessing we didn't want to use the word "resolver" again here so opt to using provider?